src/devtools/client/inspector/markup/new-markup.js

```diff
@@ -64,6 +70,44 @@ class MarkupView {
 
     // Expose the provider to let inspector.js use it in setupSidebar.
     this.provider = provider;
+
+    if (ThreadFront.currentPause) {
+      await this.onPaused();
+    } else {
+      this.onResumed();
+    }
+  }
+
+  async onPaused() {
+    this.store.dispatch(reset());
+
+    ThreadFront.ensureCurrentPause();
+    const pause = ThreadFront.currentPause;
+    if (this.selection.nodeFront && this.selection.nodeFront.pause !== pause) {
+      this.selection.setNodeFront(null);
+    }
+
+    const rootNode = await ThreadFront.getRootDOMNode();
+    await rootNode?.ensureLoaded();
+    if (!rootNode || ThreadFront.currentPause !== pause) {
+      return;
+    }
+    this.store.dispatch(newRoot(rootNode));
+
+    if (this.selection.nodeFront) {
+      this.store.dispatch(selectionChanged(this.selection, false));
+    } else {
+      const defaultNode = await rootNode.querySelector("body");
+      await defaultNode?.ensureLoaded();
+      if (defaultNode && !this.selection.nodeFront && ThreadFront.currentPause === pause) {
+        this.selection.setNodeFront(defaultNode, { reason: "navigateaway" });
+      }
+    }
+  }
```

Comment:
We don't need to do this now, but it would be nice if this business logic were in a redux action. The reason for that, is that it helps us keep state out of the code and encapsulate some of this business logic. 

see https://github.com/RecordReplay/devtools/blob/master/src/devtools/client/webconsole/actions/messages.js#L30-L44